plugins/swagger/swagger.go

```diff
@@ -0,0 +1,180 @@
+/*
+Copyright 2018 Caicloud Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package swagger
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/caicloud/nirvana"
+	"github.com/caicloud/nirvana/definition"
+	"github.com/caicloud/nirvana/service"
+)
+
+func init() {
+	// Register swagger config installer into nirvana.
+	nirvana.RegisterConfigInstaller(&swaggerInstaller{})
+}
+
+// ExternalConfigName is the external config name of swagger.
+const ExternalConfigName = "swagger"
+
+type swaggerInstaller struct{}
+
+// Name is the external config name.
+func (s *swaggerInstaller) Name() string {
+	return ExternalConfigName
+}
+
+// Install installs config to builder. You can get plugin config from nirvana config. Then
+// install/initialize what you need.
+func (s *swaggerInstaller) Install(builder service.Builder, cfg *nirvana.Config) error {
+	var err error
+	wrapper(cfg, func(c *config) {
+		for _, v := range c.config {
+			path, filePath, err := splitConfig(v)
+			if err != nil {
+				panic(err)
+			}
+			flag, err := pathExists(filePath)
+			if err != nil {
+				panic(err)
+			}
+			if !flag {
+				panic(fmt.Errorf("File %s not exists", filePath))
+			}
+
+			err = builder.AddDescriptor(definition.Descriptor{
+				Path:     path,
+				Consumes: []string{definition.MIMEAll},
+				Produces: []string{definition.MIMEAll},
+				Definitions: []definition.Definition{{
+					Method:  definition.Get,
+					Results: definition.DataErrorResults(""),
+					Function: func(ctx context.Context) ([]byte, error) {
+						b, err := ioutil.ReadFile(filePath)
+						if err != nil {
+							return nil, err
+						}
+						return b, nil
+					},
+				}},
+			})
+		}
+	})
+	return err
+}
+
+// Uninstall uninstalls stuffs after server terminating.
+func (s *swaggerInstaller) Uninstall(builder service.Builder, cfg *nirvana.Config) error {
+	return nil
+}
+
+// Disable returns a configurer to disable current plugin for a certain nirvana server.
+func Disable() nirvana.Configurer {
+	return func(c *nirvana.Config) error {
+		// Set to nil will delete plugin config from nirvana config.
+		c.Set(ExternalConfigName, nil)
+		return nil
+	}
+}
+
+type config struct {
+	config []string
+}
+
+// SetOptions returns a configurer to set swagger options.
+func SetOptions(option Option) nirvana.Configurer {
```

Comment:
Please refer to https://github.com/caicloud/nirvana/blob/master/plugins/reqlog/reqlog.go.
The plugin does not obey the convention of Nirvana.

Option is a convenient way for generating command flags/ENV. It's not for users to set plugin config.

Reply:
Also you need to provide `Default()` and `Disable()`.